core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java

```diff
@@ -65,23 +67,29 @@ protected void begin() {
     protected void processSystem() {
         for (int entityId : sortedEs) {
             entityPosition = positionComponentMapper.get(entityId);
-            entityRenderable = renderableComponentMapper.get(entityId);
-            batch.setColor(entityRenderable.r, entityRenderable.g, entityRenderable.b, entityRenderable.a);
-            batch.draw(
-                    entityRenderable.texture,
-                    entityPosition.x,
-                    entityPosition.y,
-                    entityPosition.width / 2f,
-                    entityPosition.height / 2f,
-                    entityPosition.width,
-                    entityPosition.height,
-                    entityRenderable.scaleX,
-                    entityRenderable.scaleY,
-                    entityPosition.rotation,
-                    0, 0,
-                    entityRenderable.texture.getWidth(),
-                    entityRenderable.texture.getHeight(),
-                    false, false);
+            if (renderableComponentMapper.has(entityId)) {
+                entityRenderable = renderableComponentMapper.get(entityId);
+                batch.setColor(entityRenderable.r, entityRenderable.g, entityRenderable.b, entityRenderable.a);
+                batch.draw(
+                        entityRenderable.texture,
+                        entityPosition.x,
+                        entityPosition.y,
+                        entityPosition.width / 2f,
+                        entityPosition.height / 2f,
+                        entityPosition.width,
+                        entityPosition.height,
+                        entityRenderable.scaleX,
+                        entityRenderable.scaleY,
+                        entityPosition.rotation,
+                        0, 0,
+                        entityRenderable.texture.getWidth(),
+                        entityRenderable.texture.getHeight(),
+                        false, false);
+            }
+            if (textComponentMapper.has(entityId)) {
+                Text text = textComponentMapper.get(entityId);
+                text.font.draw(batch, text.text, entityPosition.x, entityPosition.y);
+            }
```

Comment:
Might be worth looking into how we could generalize this into a single call that would delegate the actual drawing interactions to a subclass. Or if we want to keep it like this, can an entity be both a Renderable and a Text?